File: bukkit/src/main/java/dev/mardroide/bukkit/listeners/AsyncChatListener.java
package dev.mardroide.bukkit.listeners;

import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.AsyncPlayerChatEvent;

public class AsyncChatListener implements Listener {
    @EventHandler
    public void onAsyncChat(AsyncPlayerChatEvent event) {
        String message = event.getMessage();
        Player player = event.getPlayer();

        event.setFormat(ChatColor.GOLD + player.getDisplayName() + ": "+ ChatColor.GRAY + message);
    }
}

File: core/src/main/kotlin/listeners/AsyncChatListener.kt
package one.hyro.listeners

import io.papermc.paper.chat.ChatRenderer
import io.papermc.paper.event.player.AsyncChatEvent
import net.kyori.adventure.audience.Audience
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor
import org.bukkit.World
import org.bukkit.entity.Player
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

object AsyncChatListener: Listener, ChatRenderer {
    @EventHandler
    fun onAsyncChat(event: AsyncChatEvent) {
        event.renderer(this::render)
    }

    override fun render(player: Player, displayName: Component, message: Component, audience: Audience): Component {
        return displayName.color(NamedTextColor.DARK_GRAY)
            .append(Component.text(": ", NamedTextColor.DARK_GRAY))
            .append(message.color(NamedTextColor.GRAY))
    }
}